pkg/controller/controller.go

```diff
@@ -166,56 +181,87 @@ func (haproxy *HAProxyController) OnUpdate(cfg ingress.Configuration) error {
 		return nil
 	}
 
-	out, err := haproxy.reloadHaproxy()
+	reloadCmd := haproxy.reloadHaproxy(configFile)
+	out, err := reloadCmd.CombinedOutput()
 	if len(out) > 0 {
-		glog.Infof("HAProxy output:\n%v", string(out))
+		glog.Infof("HAProxy[pid=%v] output:\n%v", reloadCmd.Process.Pid, string(out))
 	}
 	return err
 }
 
 // RewriteConfigFiles safely replaces configuration files with new contents after validation
-func rewriteConfigFiles(data []byte, reloadStrategy, configFile string) error {
```

Comment:
I noticed that you are rewriting the `multibinder` and non-`multibinder` usages here, which concerns me mostly due to fear of change than anything else. Curious, did you consider the following flow instead and why would it be a bad approach?

```
func rewriteConfigFiles(data []byte, reloadStrategy, configFile string) error {
  // Determine temp file path using a timestamp
  // Write rendered template file there
  // if reload strategy == multibinder then do multibinder logic for moving it into place
  // else do non-multibinder work with config file
  // copy temp file to location for later investigation if required
  // rename file to the /etc/haproxy/haproxy.cfg
}
```

Seems to me that this approach would minimize the amount of code you would need to touch and still give you a rendered template somewhere for investigation? Also unsure about where in that flow a cleanup step would be most efficient.

Reply:
If I'm following your pseudo code, wouldn't the last step prevent each haproxy process from pointing to a unique timestamped config file? That would defeat the primary changed intended by this pull request (for the non-multibinder usage).

I agree with the sentiment to touch only the code that needs to change. I initially tried this but once I understood what it did I felt some bits of logic were confusing enough that it would be improved with what you see here. For example, I found it suprising that the non-multibinder case used the post-rename config file, while the multibinder case did not. I think this change makes the behavior in each usage more clear, but I may be wrong and am definitely open.

Thanks for taking a look.

Reply:
I see the difference in approach now. My hope was that all reload strategies would be capable of at least storing copies of their config at reload time (by copying the file off before putting it into place), but understand that is simply adding files that provide limited value. Look forward to verifying this code when the new version gets built and tagged!